src/plugins/network/network.c

```diff
@@ -50,6 +52,75 @@ int elektraNetworkAddrInfo (Key * toCheck)
 	return 0;
 }
 
+int elektraPortInfo (Key * toCheck, Key * parentKey)
+{
+	const Key * meta = keyGetMeta (toCheck, "check/port");
+	const Key * listenMeta = keyGetMeta (toCheck, "check/port/listen");
+	if (!meta && !listenMeta) return 0; /* No check to do */
+	char * endptr = NULL;
+	long portNumber = strtol (keyString (toCheck), &endptr, 10);
+	int portNumberNetworkByteOrder;
+
+	if (*endptr == '\0')
+	{
+		if (portNumber < 0 || portNumber > 65535)
+		{
+			ELEKTRA_SET_ERRORF (201, parentKey, "Port %ld on key %s was not within 0 - 65535", portNumber, keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = htons (portNumber);
+	}
+	else
+	{
+		struct servent * service;
+		service = getservbyname (keyString (toCheck), NULL); // NULL means we accept both tcp and udp
+		if (service == NULL)
+		{
+			ELEKTRA_SET_ERRORF (202, parentKey, "Could not find service with name %s on key %s", keyString (toCheck),
+					    keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = service->s_port;
+	}
+
+	// Check if we can connect to it
+	if (!listenMeta) return 0; /* No check to do */
+
+	char * hostname = "localhost";
+
+	int sockfd;
+	struct sockaddr_in serv_addr;
+	struct hostent * server;
+	sockfd = socket (AF_INET, SOCK_STREAM, 0);
+	if (sockfd < 0)
+	{
+		ELEKTRA_SET_ERROR (204, parentKey, "Could not open a socket");
+	}
+
+	server = gethostbyname (hostname);
+	if (server == NULL)
+	{
+		ELEKTRA_SET_ERRORF (204, parentKey, "Could not connect to %s: No such host", hostname);
+		return -1;
+	}
+
+	bzero ((char *) &serv_addr, sizeof (serv_addr));
+	serv_addr.sin_family = AF_INET;
+	bcopy ((char *) server->h_addr, (char *) &serv_addr.sin_addr.s_addr, server->h_length);
+
+	serv_addr.sin_port = (in_port_t) portNumberNetworkByteOrder;
+	if (connect (sockfd, (struct sockaddr *) &serv_addr, sizeof (serv_addr)) == 0)
```

Comment:
Is there no other way? If you connect to it, you might cause errors in later attempts to connect ("port still in use" errors)

Reply:
I read an article about this: [link](https://stackoverflow.com/questions/10294515/how-do-i-find-in-c-that-a-port-is-free-to-use)
Is there a notable difference between bind and connect?

Reply:
Parsing /proc/net/tcp is also possible. Comparing different ways for how to do validations is part of your work.

Reply:
After doing some research there are 2 possibilities:

1. Parse /proc/net/tcp as you suggested
2. Call bind/ connect and see if the port is in use

Doing parsing is very slow and as [this comment](https://stackoverflow.com/a/10311153) suggests is no superior anyway compared to bind. As a result I will stick to the bind approach.

Reply:
After digging another time into the internet I was not able to find any other possibility other than trying to connect or parsing proc/net/tcp. Since the first possibility is superior (same disadvantages but faster) I will keep the code as it is